homeassistant/components/google_assistant/trait.py

```diff
@@ -327,6 +336,75 @@ async def execute(self, command, data, params, challenge):
         }
 
 
+@register_trait
+class ObjectDetection(_Trait):
+    """Trait to object detection.
+
+    https://developers.google.com/actions/smarthome/traits/objectdetection
+    """
+
+    name = TRAIT_OBJECTDETECTION
+    commands = []
+
+    @staticmethod
+    def supported(domain, features, device_class, _) -> bool:
+        """Test if state is supported."""
+        return domain == event.DOMAIN
+
+    def sync_attributes(self):
+        """Return ObjectDetection attributes for a sync request."""
+        return {}
+
+    def sync_options(self) -> dict[str, Any]:
+        """Add options for the sync request."""
+        return {"notificationSupportedByAgent": True}
+
+    def query_attributes(self):
+        """Return ObjectDetection query attributes."""
+
+        async def _async_send_notification(time_stamp: int) -> None:
+            payload = {
+                "devices": {
+                    "notifications": {
+                        self.state.entity_id: {
+                            "ObjectDetection": {
+                                "priority": 0,
+                                "detectionTimestamp": time_stamp,
+                            }
+                        }
+                    }
+                }
+            }
+            event_id = self.state.attributes.get("event_type")
```

Comment:
Event id is not a static type of event. Its UUID that is unique every time a event is triggered.

Reply:
Ps. You can generate a random number to test. But we need a better solution to get it from the context of the event that triggered the state change layer.